mkosi/resources/mkosi.md

```diff
@@ -2023,6 +2029,13 @@ current working directory. The following scripts are supported:
   artifacts from `SplitArtifacts=yes` or RPMs built in a build script).
   Note that this script does not use the tools tree even if one is configured.
 
+* If **`mkosi.postoutput`** (`PostOutputScripts=`) exists, it is executed right
+  after all the output files have been generated, before they are finally
+  moved into the output directory. A post-output script can be used to
+  introspect the generated output files to modify any `BuildSource=` mount
+  (e.g. writing a custom `SHA256SUMS` file). Note that this script does not use
```

Comment:
Typo: `BuildSource=` → `BuildSources=`.

I'm not entirely sure I understand how the `BuildSources=` come in here, since the example is about `SHA256SUMS` and this seems a bit incongruent, i.e. If this is about the sums and signing, why would I want to modify the build sources at this point? On that note: signing should probably be mentioned here as well.

Reply:
With the output dir mounted read-only that is somewhat the only option if you want to write some custom output persistently.

I guess the `SHA256SUMS` example was a bit vague and "me usecase" specific, but its to *not* use the integrated `SHA256SUMS` file as that always gets removed during clean, only contains the info for a specific image and is prepended with the image name.

My intended usecase is to append the sums to a common `SHA256SUMS` file (and only remove the lines for a specific image in `mkosi.clean`) so that it can be consumed by `sd-sysupdate`.

Reply:
> With the output dir mounted read-only that is somewhat the only option if you want to write some custom output persistently.

> Currently I have it that the /work/out ($OUTPUTDIR) is mounted read-only, as I would expect that the output shouldn't be modified after it was built, hence only introspect the output.

Mhm, I see the point about the outputs not being changed after building, but writing a different `SHA256SUMS` is, I think, still producing an output and staging that in `BuildSources=` to persist it, does not seem ideal to me, one of the reasons being interaction with the ephemeral build sources option.

Reply:
> one of the reasons being interaction with the ephemeral build sources option.

This is similarly handled as with the `mkosi.sync` scripts, where the build sources are always mounted non-ephemeral https://github.com/NekkoDroid/mkosi/blob/c087b9cb1190ca754e2126596c4dda745c178eaf/mkosi/__init__.py#L856. If we want to keep it this way I probably should mention this in the docs.

Anyway, my usecase doesn't exactly hinge on the output being read-only or not, but I do need some way to be able to access 1 directory up from the output dir (on host) without it being ephemeral. And this would currently be the main way I'd see doing this without a rework how mkosi handles its `SHA256SUMS` file.

Reply:
Build sources definitely need to be ephemeral in the postoutput scripts if the setting is used. These scripts are specifically intended to operate on the outputs, not on the sources so it does not make sense for the sources to not be ephemeral when running output scripts.